src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs

```diff
@@ -147,38 +149,51 @@ internal void ModifyManifest(string basePath)
 
         internal void SetupSymbolsUpload(string unityProjectPath, string gradleProjectPath)
         {
-            var options = _getOptions();
-            var logger = options?.DiagnosticLogger ?? new UnityLogger(new SentryUnityOptions());
+            var disableSymbolsUpload = false;
+            var logger = _options?.DiagnosticLogger ?? new UnityLogger(new SentryUnityOptions());
+            var symbolsUpload = new DebugSymbolUpload(logger, unityProjectPath, gradleProjectPath,
+                EditorUserBuildSettings.exportAsGoogleAndroidProject);
 
-            if (_scriptingImplementation != ScriptingImplementation.IL2CPP)
+            if (_options is null)
             {
-                logger.LogDebug("Automated symbols upload requires the IL2CPP scripting backend.");
-                return;
+                disableSymbolsUpload = true;
+            }
+            else if (!_options.Enabled)
+            {
+                disableSymbolsUpload = true;
+            }
+            else if (!_options.AndroidNativeSupportEnabled)
+            {
+                disableSymbolsUpload = true;
             }
 
-            var symbolsUpload = new DebugSymbolUpload(logger, unityProjectPath, gradleProjectPath, EditorUserBuildSettings.exportAsGoogleAndroidProject);
-
-            var sentryCliOptions = _getSentryCliOptions();
-            if (sentryCliOptions is null)
+            if (_sentryCliOptions is null)
             {
                 logger.LogWarning("Failed to load sentry-cli options.");
-                symbolsUpload.RemoveUploadFromGradleFile();
-
-                return;
+                disableSymbolsUpload = true;
+            }
+            else if (!_sentryCliOptions.IsValid(logger, _isDevelopmentBuild))
+            {
+                disableSymbolsUpload = true;
```

Comment:
maybe a log so that users get at least some feedback that symbols are not uploaded and (kinda) why?

Reply:
The "why" logs come from within `IsValid`.